File: CreationalPatterns/Builder/TypeScript/CarManualBuilder.ts
import { Builder } from './Builder';
import { Manual } from './Manual';

export class CarManualBuilder implements Builder {
    private manual: Manual = new Manual();

    reset(): void {
        this.manual = new Manual();
    }
    setSeats(seats: number): void {
        this.manual.seats = seats;
    }
    setEngine(engine: string): void {
        this.manual.engine = engine;
    }
    setTripComputer(hasTripComputer: boolean): void {
        this.manual.tripComputer = hasTripComputer;
    }
    setGPS(hasGPS: boolean): void {
        this.manual.gps = hasGPS;
    }
    getResult(): Manual {
        return this.manual;
    }
}

File: CreationalPatterns/Builder/CSharp/CarManualBuilder.cs
namespace Builder
{
    public class CarManualBuilder : IBuilder
    {
        private Manual _manual = new Manual();

        public void Reset()
        {
            _manual = new Manual();
        }

        public void SetSeats(int number)
        {
            _manual.Seats = number;
        }

        public void SetEngine(string engine)
        {
            _manual.Engine = engine;
        }

        public void SetTripComputer(bool hasTripComputer)
        {
            _manual.TripComputer = hasTripComputer;
        }

        public void SetGPS(bool hasGPS)
        {
            _manual.GPS = hasGPS;
        }

        public Manual GetResult()
        {
            return _manual;
        }
    }
}

